src/solvers/dg/2d/dg.jl

```diff
@@ -1849,47 +1907,37 @@ end
 # Calculate and store boundary flux across domain boundaries
 #NOTE: Do we need to dispatch on have_nonconservative_terms(dg.equations)?
 calc_boundary_flux!(dg::Dg2D, time) = calc_boundary_flux!(dg.elements.surface_flux_values,
-                                                          dg, time)
-function calc_boundary_flux!(surface_flux_values, dg::Dg2D, time)
+                                                          dg, time,
+                                                          dg.boundary_conditions...)
+function calc_boundary_flux!(surface_flux_values, dg::Dg2D, time, boundary_conditions...)
   @unpack surface_flux_function = dg
   @unpack u, neighbor_ids, neighbor_sides, node_coordinates, orientations = dg.boundaries
 
-  Threads.@threads for b in 1:dg.n_boundaries
-    # Fill outer boundary state
-    # FIXME: This should be replaced by a proper boundary condition
-    for i in 1:nnodes(dg)
-      @views u[3 - neighbor_sides[b], :, i, b] .= dg.initial_conditions(
-        node_coordinates[:, i, b], time, equations(dg))
-    end
-
-    # Get neighboring element
-    neighbor_id = neighbor_ids[b]
+  # Calculate indices
+  lasts = accumulate(+, dg.n_boundaries_per_direction)
+  firsts = lasts - dg.n_boundaries_per_direction + 1
 
-    # Determine boundary direction with respect to elements:
-    # orientation = 1: left -> 2, right -> 1
-    # orientation = 2: left -> 4, right -> 3
-    if orientations[b] == 1 # Boundary in x-direction
-      if neighbor_sides[b] == 1 # Element is on the left, boundary on the right
-        direction = 2
-      else # Element is on the right, boundary on the left
-        direction = 1
-      end
-    else # Boundary in y-direction
-      if neighbor_sides[b] == 1 # Element is below, boundary is above
-        direction = 4
-      else # Element is above, boundary is below
-        direction = 3
-      end
-    end
+  for direction in 1:4
+    Threads.@threads for b in firsts[direction]:lasts[direction]
```

Comment:
Maybe we should introduce a function barrier between these lines to make the choice of `boundary_conditions[direction]` below type-stable?

Reply:
Aha! Didn't read this up there, sorry. Let me try this out...

Reply:
Thank you for the suggestion! Fixed in d04aa62.